Backend/utils/cronJobRecommendation.js

```diff
@@ -63,7 +63,31 @@ const tagWeights = {
 };
 
 async function calculateRecommendations() {
-    const songs = await prisma.song.findMany({
+    const MILLISECONDS_PER_SECOND = 1000;
+    const SECONDS_PER_MINUTE = 60;
+    const MINUTES_PER_HOUR = 60;
+    const HOURS_AGO = 3;
+    const threeHoursAgo = new Date(Date.now() - HOURS_AGO * MINUTES_PER_HOUR * SECONDS_PER_MINUTE * MILLISECONDS_PER_SECOND);
+
+    // Get songs with tags updated in the last 3 hours
+    const recentSongs = await prisma.song.findMany({
+        where: {
+            updatedAt: {
+                gte: threeHoursAgo
+            }
+        },
+        include: {
+            user: true
+        }
+    });
+
+    const uniqueUserIds = recentSongs.reduce((acc, song) => {
```

Comment:
Can you rename uniqueUserIds to uniqueUsersWithRecentTags? Since this is not a set of all users, just users that have created a new tag in the last 3 hours